plainly-plugin/src/ui/components/form/UploadForm.tsx

```diff
@@ -0,0 +1,157 @@
+import FormData from 'form-data';
+import { useState } from 'react';
+import {
+  collectFiles,
+  removeFolder,
+  retrieveSettings,
+  zip,
+} from '../../../node';
+import { post } from '../../../node/request';
+import Button from '../common/Button';
+import Description from '../typography/Description';
+import Label from '../typography/Label';
+import PageHeading from '../typography/PageHeading';
+
+import fs from 'fs';
+import path from 'path';
+import { settingsDirectory } from '../../../node/constants';
+import { useNotification } from '../../hooks/useNotification';
+import Notification from '../common/Notification';
+
+export default function UploadForm() {
+  const [inputs, setInputs] = useState<{
+    projectName?: string;
+    description?: string;
+    tags?: string[];
+  }>({});
+  const [loading, setLoading] = useState(false);
+  const { notification, notifySuccess, notifyError, clearNotification } =
+    useNotification();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setLoading(true);
+
+    try {
+      const { apiKey } = await retrieveSettings();
```

Comment:
* why not useSettings, that's why we created that?
* useSettings can also provide maybe a function to get the apikey, no matter if the key is encryped or not (later on)